lib/block_view.go

```diff
@@ -3826,11 +3786,19 @@ func (bav *UtxoView) GetPostEntryForPostHash(postHash *BlockHash) *PostEntry {
 	// If we get here it means no value exists in our in-memory map. In this case,
 	// defer to the db. If a mapping exists in the db, return it. If not, return
 	// nil.
-	dbPostEntry := DBGetPostEntryByPostHash(bav.Handle, postHash)
-	if dbPostEntry != nil {
-		bav._setPostEntryMappings(dbPostEntry)
+	if bav.Postgres != nil {
+		post := bav.Postgres.GetPost(postHash)
+		if post != nil {
+			return bav.setPostMappings(post)
```

Comment:
Nit: We should be consistent about what layer of the stack PG* structs are allowed. I think you shouldn't write set*Mappings functions to take PG* structs. Instead I would cast it first to keep taht interface clean and consistent.